### DISCUSSION TITLE
Graphhopper info

### DISCUSSION THREAD

**Discussion by moiatto** [mapsforge/vtm#1122] [Q&A] [answered]
Hi devemux86,
I know this isn't the right place to ask this, but you know enough to be able to answer.
I'm starting to study the graphhopper offline routing library v0.13 because mapforge server stores compatible graphs for this version.
The only limit I founded is about multi areas routing.
I have read this [topic](https://github.com/graphhopper/graphhopper/issues/293) and I want understand if this is a limit of this version 0.13 or is a structural problem not solved even in subsequent versions.
Is there a not too complex trick to get around this limit? Alternatively, what do you recommend for offline routing (which has no regional limits)?

**Comment by devemux86** [comment, marked as answer]
@moiatto 

Mapsforge server provides GraphHopper 1.0 graphs.
(last version supporting all Android versions: https://github.com/graphhopper/graphhopper/issues/1940)

Multi region offline routing is not supported by GraphHopper.

---

**BRouter** is an offline router with customizable profiles and many options (without region limits):
https://github.com/abrensch/brouter/

**Cruiser** provides detailed offline routing with BRouter: https://github.com/devemux86/cruiser
